lang/syntax/src/trees/generic/def.rs

```diff
@@ -194,6 +215,13 @@ pub struct Let<P: Phase> {
     pub body: Rc<Exp<P>>,
 }
 
+impl<P: Phase> Let<P> {
+    /// Returns whether the declaration is the "main" expression of the module.
+    pub fn is_main(&self) -> bool {
+        self.name == "main" && self.params.is_empty()
```

Comment:
We should probably forbid parameters on let bindings called `main`  during lowering.

Reply:
Reasoning: otherwise it is confusing for users because they will see the error message "main not found" but they have a `main` expression with a different signature.